### PR TITLE
[Rust] Bump to Rust v1.64.0

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
- [x] Build script tested locally
- [ ] After this PR, follow up with BinaryBuilder `Artifacts.toml` update

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM julia:1.7
 
-RUN apt-get update && apt-get install -y xz-utils bzip2 sudo git unzip
```
Comment (giordano) [inline .devcontainer/Dockerfile:3]: This change seems unrelated

Comment (jeremiahpslewis) [reply .devcontainer/Dockerfile:3]: On second thought, think this solves it a bit more elegantly: https://github.com/JuliaPackaging/Yggdrasil/pull/5541

**Comment by giordano** [comment]
For the record, it appears that it's Rustup v1.25.1 which completely screws up our toolchain.  Downgrading to Rustup v1.24.3 I get again a working compiler.
